src/Viewer/components/Monaco/MonacoInstance.js

```diff
@@ -265,29 +186,79 @@ function MonacoInstance ({
         }
     };
 
-    useEffect(() => {
-        window.addEventListener("keypress", handleKeyDown);
-        return () => {
-            window.removeEventListener("keypress", handleKeyDown);
-        };
-    }, []);
+    const initMonacoEditor = () => {
+        if (null === editorRef.current) {
```

Comment:
Henry and I briefly discussed this yesterday and I suggested the single exit-point approach.

Would it be better to keep a single `return` statement to simply the flow? Or do we permit early return statements for failed parameter validations at the top of functions?

Reply:
We do permit early return statements for failed parameter validations. There is also a recommended practice (I think it comes from the early Linux devs) of using returns (and in their case, gotos) to avoid a lot of nested branches.

Reply:
Right, otherwise we can maintain a status code and keep checking that status code in the middle to avoid nested-if blocks, which is particularly helpful for functions that return a status code. 

Let me put the early return rule into our coding guidelines. :)

Reply:
Of course, it isn't a hard and fast rule and sometimes branching is more readable than early returns.